client/state/terms/actions.js

```diff
@@ -45,7 +45,6 @@ export function addTerm( siteId, taxonomy, term ) {
 					dispatch( receiveTerm( siteId, taxonomy, data ) );
 					return data;
 				},
-				() => Promise.resolve() // Silently ignore failure so we can proceed to remove temporary
```

Comment:
Also, we may need to run prettier on it after the changes, since if we remove the second argument of `.then()`, prettier will want to reformat it (see how the next `.then()` is formatted).